File: benchmark/string_equals.lua
local start = os.clock()

local count = 0
for i = 1, 1000000 do
  if ("abc" == "abc") then count = count + 1 end
  if ("a slightly longer string" ==
      "a slightly longer string") then count = count + 1 end
  if ("a significantly longer string but still not overwhelmingly long string" ==
      "a significantly longer string but still not overwhelmingly long string") then count = count + 1 end

  if ("" == "abc") then count = count + 1 end
  if ("abc" == "abcd") then count = count + 1 end
  if ("changed one character" == "changed !ne character") then count = count + 1 end
  if ("123" == 123) then count = count + 1 end
  if ("a slightly longer string" ==
      "a slightly longer string!") then count = count + 1 end
  if ("a slightly longer string" ==
      "a slightly longer strinh") then count = count + 1 end
  if ("a significantly longer string but still not overwhelmingly long string" ==
      "another") then count = count + 1 end
end

print(count)
print("elapsed: "..(os.clock() - start))

File: benchmark/string_equals.js
var start = new Date().getTime();

var count = 0;
for (var i=0; i<1000000; i++) {
  if ("abc" == "abc") count = count + 1;
  if ("a slightly longer string" ==
      "a slightly longer string") count = count + 1;
  if ("a significantly longer string but still not overwhelmingly long string" ==
      "a significantly longer string but still not overwhelmingly long string") count = count + 1;

  if ("" == "abc") count = count + 1;
  if ("abc" == "abcd") count = count + 1;
  if ("changed one character" == "changed !ne character") count = count + 1;
  if ("123" == 123) count = count + 1;
  if ("a slightly longer string" ==
      "a slightly longer string!") count = count + 1;
  if ("a slightly longer string" ==
      "a slightly longer strinh") count = count + 1;
  if ("a significantly longer string but still not overwhelmingly long string" ==
      "another") count = count + 1;
}

start = (new Date().getTime() -start)/1000.0;
console.log(count);
console.log("Elapsed: "+start);
